doc/netplan-tutorial.md

```diff
@@ -0,0 +1,560 @@
+
+```bash
+$ snap install lxd
+$ lxd init --minimal
+```
+
+For some of the exercises you will need a second network interface in your virtual machine. Run the command below to create a new network in LXD:
+
+```bash
+lxc network create netplanbr0 --type=bridge
+```
+
+You should see the output below:
+
+```
+Network netplanbr0 created
+```
+
+After executing the commands above you will have a usable LXD installation with a working network bridge.
+
+Now create a virtual machine called `netplan-lab0`:
+
+```bash
+lxc init --vm ubuntu:22.04 netplan-lab0
+```
+
+You should see the out below:
+
+```
+Creating netplan-lab0
+```
+
+The new VM will have one network interface attached to the default LXD bridge.
+
+Now attach a second network interface to the VM and associate it to the LXD network you created earlier:
+
+```bash
+lxc network attach netplanbr0 netplan-lab0 eth1
+```
+
+And start your new VM:
+
+```bash
+lxc start netplan-lab0
+```
+
+Access your new VM using `lxc exec`:
+
+```bash
+lxc exec netplan-lab0 bash
+```
+
+You should now have a root shell inside your VM:
+
+```bash
+root@netplan-lab0:~#
+```
+
+If you see the command prompt above, congratulations, you got into your LXD virtual machine!
+
+Run the command `ip link` to show your network interfaces:
+
+```bash
+ip link
+```
+
+You should see an output similar to the below:
+
+```
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+2: enp5s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
+    link/ether 00:16:3e:27:45:9d brd ff:ff:ff:ff:ff:ff
+3: enp6s0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
+    link/ether 00:16:3e:7b:68:9d brd ff:ff:ff:ff:ff:ff
+```
+
+In this case, `enp5s0` is the primary interface connected to the default LXD network and `enp6s0` is the second interface you added connected to your custom network.
+
+
+Now, let's start with a simple exercise.
+
+## Running netplan for the first time
+
+Start by typing the command `netplan` in your shell:
+
+```bash
+netplan
+```
+
+You will see the output below
+
+```
+You need to specify a command
+usage: /usr/sbin/netplan  [-h] [--debug]  ...
+
+Network configuration in YAML
+
+options:
+  -h, --help  show this help message and exit
+  --debug     Enable debug messages
+
+Available commands:
+
+    help      Show this help message
+    apply     Apply current netplan config to running system
+    generate  Generate backend specific configuration files from /etc/netplan/*.yaml
+    get       Get a setting by specifying a nested key like "ethernets.eth0.addresses", or "all"
+    info      Show available features
+    ip        Retrieve IP information from the system
+    set       Add new setting by specifying a dotted key=value pair like ethernets.eth0.dhcp4=true
```

Comment:
I'm missing the new `netplan status` here (and in the overall tutorial).

Reply:
Yeah I almost included it (I added a note to the PR description). As we have plans to backport Netplan 0.106 I guess I should add it.